test/cppsuite/test_harness/runtime_monitor.h

```diff
@@ -109,13 +110,74 @@ class cache_limit_statistic : public statistic {
     int64_t limit;
 };
 
+class db_size_statistic : public statistic {
+    public:
+    db_size_statistic(configuration *config, std::vector<std::string> &&file_names)
+        : statistic(config), _file_names(std::move(file_names))
+    {
+        _limit = config->get_int(LIMIT);
+    }
+    virtual ~db_size_statistic() = default;
+
+    /* Don't need the stat cursor for this. */
+    void
+    check(WT_CURSOR *) override final
+    {
+#ifndef _WIN32
+        size_t db_size = 0;
+        for (const auto &name : _file_names) {
+            struct stat sb;
+            if (stat(name.c_str(), &sb) == 0) {
+                db_size += sb.st_size;
+                debug_print(name + " was " + std::to_string(sb.st_size) + " bytes", DEBUG_TRACE);
+            } else
+                /* The only good reason for this to fail is if the file hasn't been created yet. */
+                testutil_assert(errno == ENOENT);
+        }
+        debug_print("Current database size is " + std::to_string(db_size) + " bytes", DEBUG_TRACE);
+        if (db_size > _limit) {
+            std::string error_string =
+              "runtime_monitor: Database size limit exceeded during test! Limit: " +
+              std::to_string(_limit) + " db size: " + std::to_string(db_size);
+            debug_print(error_string, DEBUG_ERROR);
+            testutil_assert(db_size <= _limit);
+        }
+#else
+        static_cast<void>(_file_names);
+        static_cast<void>(_limit);
+#endif
+    }
+
+    private:
+    std::vector<std::string> _file_names;
+    int64_t _limit;
+};
+
+static std::vector<std::string>
+collection_identifiers_to_file_names(const std::vector<std::string> &identifiers)
+{
+    /*
+     * At the moment, the home directory isn't configurable so this is fine. We'll need to pipe this
+     * through when it is configurable.
+     */
+    std::vector<std::string> file_names;
+    for (const auto &identifier : identifiers)
+        file_names.push_back(std::string(DEFAULT_DIR) + "/" + identifier + ".wt");
+    file_names.push_back(std::string(DEFAULT_DIR) + "/" + WT_HS_FILE);
+    file_names.push_back(std::string(DEFAULT_DIR) + "/" + WT_METAFILE);
+    return file_names;
+}
+
 /*
  * The runtime monitor class is designed to track various statistics or other runtime signals
  * relevant to the given workload.
  */
 class runtime_monitor : public component {
     public:
-    runtime_monitor(configuration *config) : component("runtime_monitor", config) {}
+    runtime_monitor(configuration *config, database_operation *db_operation)
```

Comment:
Instead of giving the database operation can we pass it the database model, which contains a list of collection names?

Reply:
Done